src/serialization/container.h

```diff
@@ -52,13 +52,32 @@ namespace serialization
       static constexpr const bool previously_varint = std::is_same<uint64_t, T>() || std::is_same<uint32_t, T>();
 
       if (!previously_varint && ar.varint_bug_backward_compatibility_enabled() && !typename Archive::is_saving())
-        return ::do_serialize(ar, e);
+        return do_serialize(ar, e);
       ar.serialize_varint(e);
       return true;
     }
 
+    //! @brief Add an element to a container, inserting at the back if applicable.
+    template <class Container>
+    auto do_add(Container &c, typename Container::value_type &&e) -> decltype(c.emplace_back(e), void())
```

Comment:
I commented on this on another review, and you rebuked, but I'll try again!

Why not just make these pass-through? `emplace_back()` returns void anyway, and its not terribly to forward the return-call (if there is one).

Reply:
I guess I don't care as much about the return type of these functions since they don't need to be chained together like `do_serialize()`. It would make it less verbose anyways, so I think this way is better. FWIW, `emplace_back()` only returns `void` before C++17 and `emplace()` returns `std::pair<iterator, bool>`.